common/data_refinery_common/models/models.py

```diff
@@ -420,6 +422,27 @@ def pretty_platforms(self):
     def processed_samples(self):
         return list([sample.accession_code for sample in self.samples.all() if sample.is_processed == True])
 
+    @property
+    def downloadable_samples(self):
+        """
+        Returns the accession codes of the downloadable samples in this experiment.
+        This is indexed on elastic search and used to count the number of samples
+        on the filters.
+        """
+        # I couldn't not find a way to get calculate the samples that have organisms with QN
+        # targets in a single DB query. Since the QN Targets are `ComputationalResultAnnotation`
+        # that are not associated with the Organisms.
+        # This method will be called for each experiment when the ES index is being created
+        # This cache is to ensure that we don't repeat the query to calculate the organisms with 
+        # qn targets multiple times.
+        qn_organisms = cache.get('qn_organisms')
+        if qn_organisms == None:
+            qn_organisms = Organism.get_objects_with_qn_targets()
+            cache.set('qn_organisms', qn_organisms, 30*60) # save value for 1h
+
+        return list(self.samples.filter(is_processed=True, organism__in=qn_organisms)\
+                                      .values_list('accession_code', flat=True))
```

Comment:
Rather than caching a list of organisms that have QN targets... what if we just added a `most_recent_qn_result` field to the organisms table? When we create a new QN target we can update that and then when we need a QN target for a given organism we won't have to look through all the computational results to find it, it'll just be on the organism object? This would of course involve modifying the QN target code to update that as they're generated and a migration to populate that field for all existing QN targets.